functions/Remove-SqlOrphanUser.ps1

```diff
@@ -181,49 +186,150 @@ Will remove from all databases the user OrphanUser EVEN if exists their matching
-                                                                                   $_.IsLocked -eq $False -and 
+                                #Need to validate Existing Login because the call does not came from Repair-SqlOrphanUser
+                                $ExistLogin = $server.logins | Where-Object {$_.Isdisabled -eq $False -and
+                                                                                   $_.IsSystemObject -eq $False -and
+                                                                                   $_.IsLocked -eq $False -and
                                                                                    $_.Name -eq $User.Name }
                             }
 
-                            if ($ExistLogin)
+                            #Schemas only appears on SQL Server 2005 (v9.0)
+                            if ($server.versionMajor -gt 8)
                             {
-                                if ($Force)
+                                
+                                #Validate if user owns any schema
+                                $Schemas = @()
+
+                                $Schemas = $db.Schemas | Where-Object {$_.Owner -eq $User.Name}
+
+                                if(@($Schemas).Count -gt 0)
                                 {
-                                    if ($Pscmdlet.ShouldProcess($db.Name, "Dropping user '$($User.Name)' using -Force"))
-				                    {
-                                        $server.Databases[$db.Name].ExecuteNonQuery($query) | Out-Null
-                                        Write-Output "User '$($User.Name)' was dropped. -Force parameter was used!"
+                                    Write-Output "User '$($User.Name)' owns one or more schemas."
+
+                                    $AlterSchemaOwner = ""
+                                    $DropSchema = ""
+
+                                    foreach ($sch in $Schemas)
+                                    {
+                                        $NumberObjects = ($db.EnumObjects() | Where-Object {$_.Schema -eq $sch.Name} | Measure-Object).Count
+
+                                        if ($NumberObjects -gt 0)
+                                        {
+                                            if ($Force)
+                                            {
+                                                Write-Verbose "Parameter -Force was used! The schema '$($sch.Name)' have $NumberObjects underlying objects. We will change schema owner to 'dbo' and drop the user."
+
+                                                if ($Pscmdlet.ShouldProcess($db.Name, "Changing schema '$($sch.Name)' owner to 'dbo'. -Force used."))
+				                                {
+                                                    $AlterSchemaOwner += "ALTER AUTHORIZATION ON SCHEMA::[$($sch.Name)] TO [dbo]`r`n"
+
+                                                    $results += [pscustomobject]@{
```

Comment:
If no further manipulation of the object is needed, there is no reason to collect it in $results,
just put it in the output right away